sdk/core/core-http/src/xhrHttpClient.ts

```diff
@@ -42,7 +42,8 @@ export class XhrHttpClient implements HttpClient {
     if (request.formData) {
       const formData = request.formData;
       const requestForm = new FormData();
-      const appendFormValue = (key: string, value: any) => {
+      const appendFormValue = (key: string, value: any): void => {
+        // eslint-disable-next-line no-prototype-builtins
         if (value && value.hasOwnProperty("value") && value.hasOwnProperty("options")) {
```

Comment:
same pondering about if we should trust `value` to have a proper prototype

Reply:
I mean, I do want to give this some polishing, after seeing so many `any`s. Should we do it in this PR?

Reply:
best effort

Reply:
`best effort` I didn't understand 😵

As per Ramya here: https://github.com/Azure/azure-sdk-for-js/pull/8687#discussion_r422755318 I'll log a new issue to clean these up.

Reply:
Here's the issue: https://github.com/Azure/azure-sdk-for-js/issues/9674